library/spdm_requester_lib/libspdm_req_get_certificate.c

```diff
@@ -61,39 +52,77 @@ typedef struct {
  * @retval LIBSPDM_STATUS_CRYPTO_ERROR
  *         A generic cryptography error occurred.
  **/
-static libspdm_return_t libspdm_try_get_certificate(libspdm_context_t *spdm_context,
-                                                    const uint32_t *session_id,
-                                                    uint8_t slot_id,
-                                                    uint16_t length,
-                                                    size_t *cert_chain_size,
-                                                    void *cert_chain,
-                                                    const void **trust_anchor,
-                                                    size_t *trust_anchor_size)
+static libspdm_return_t libspdm_try_get_large_certificate(libspdm_context_t *spdm_context,
+                                                          const uint32_t *session_id,
+                                                          uint8_t slot_id,
+                                                          uint32_t length,
+                                                          size_t *cert_chain_size,
+                                                          void *cert_chain,
+                                                          const void **trust_anchor,
+                                                          size_t *trust_anchor_size)
 {
     bool result;
     libspdm_return_t status;
-    spdm_get_certificate_request_t *spdm_request;
+    spdm_get_certificate_large_request_t *spdm_request;
     size_t spdm_request_size;
-    libspdm_certificate_response_max_t *spdm_response;
+    spdm_certificate_large_response_t *spdm_response;
     size_t spdm_response_size;
-    uint16_t total_responder_cert_chain_buffer_length;
+    uint32_t total_responder_cert_chain_buffer_length;
     size_t cert_chain_capacity;
     size_t cert_chain_size_internal;
-    uint16_t remainder_length;
+    uint32_t remainder_length;
     uint8_t *message;
     size_t message_size;
     size_t transport_header_size;
     libspdm_session_info_t *session_info;
     libspdm_session_state_t session_state;
     bool chunk_enabled;
     uint8_t cert_model;
+    uint32_t req_msg_length;
+    uint32_t req_msg_offset;
+    uint32_t rsp_msg_portion_length;
+    uint32_t rsp_msg_remainder_length;
+    bool use_large_cert_chain;
+    uint32_t req_msg_header_size;
+    uint32_t rsp_msg_header_size;
+    uint32_t max_cert_chain_size;
 
     /* -=[Check Parameters Phase]=- */
     LIBSPDM_ASSERT(slot_id < SPDM_MAX_SLOT_COUNT);
     LIBSPDM_ASSERT(cert_chain_size != NULL);
     LIBSPDM_ASSERT(*cert_chain_size > 0);
     LIBSPDM_ASSERT(cert_chain != NULL);
 
+    if ((length > 0xFFFF) &&
```

Comment:
Use `SPDM_MAX_CERTIFICATE_CHAIN_SIZE*` instead of hardcoded values.

Reply:
fixed